src/_pytest/python_api.py

```diff
@@ -672,6 +668,7 @@ def raises(expected_exception, *args, **kwargs):
             raise TypeError(msg)
         return RaisesContext(expected_exception, message, match_expr)
     elif isinstance(args[0], str):
+        warnings.warn(RAISES_EXEC, stacklevel=2)
```

Comment:
Just an idea, how about if instead of a static warning, we suggest the correction already? Seems trivial in this case:

```
raises(..., 'code(as_a_string)') is deprecated, use the context manager form or use `exec()` directly:
    with raises(RuntimeError):
        int(s)   
```

?

Reply:
If you think the message will be too big, let's add a link to the `deprecations.rst` docs then.

Reply:
dedent + indent shouldn't be too hard, I'll give it a try

though, reproducing the exception isn't necessarily straightforward

```python
exc_tp = foo.MyError
pytest.raises(exc_tp, "bar()")
```

would likely get spit out in the messaging as:

```python
    with raises(MyError):
        bar()
```